repository/mock_repo.go

```diff
@@ -230,6 +231,11 @@ func (r *mockRepoForTest) HasRef(ref string) (bool, error) {
 	return true, nil
 }
 
+// HasObject reports whether or not the repo contains an object with the given hash
+func (r *mockRepoForTest) HasObject(hash string) (bool, error) {
+	return false, errors.New("Not implemented")
```

Comment:
nit: Kinda strange using the mix of `fmt.errorf` without params and `errors.New`. Is there a difference I'm unaware of?

Reply:
Using `fmt.Errorf` in those cases is wrong. In fact, `fmt.Errorf` will simply wind up calling `errors.New` in those cases.

I would say that I should go through and fix all of the instances of `fmt.Errorf("not implemented")` to use `errors.New` instead, but what I should really do is just implement them and tests that use them.